execution_environments/how_to_build_and_test_first_ee.rst

```diff
@@ -0,0 +1,156 @@
+.. code-block:: bash
+
+  $ mkdir postgresql_ee && cd postgresql_ee
+
+2. Create the ``requirements.yml`` file which will specify collections you want to include in your EE.
+Tell it that you want the ``community.postgresql`` collection:
+
+.. code-block:: yaml
+
+  ---
+  collections:
+    - name: community.postgresql
+
+3. Create the ``requirements.txt`` file which will list Python dependencies you want to include.
+In this how-to, you want the ``psycopg2-binary`` package:
+
+.. code-block:: bash
+
+  psycopg2-binary
+
+4. Create the ``execution-environment.yml`` definition file that specifies which content you want to include.
+List the files you created at previous steps:
+
+.. code-block:: yaml
+
+  ---
+  version: 3
+
+  dependencies:
+    galaxy: requirements.yml
+    python: requirements.txt
+
+Read more about the EE definition file format in the `Ansible Builder documentation <https://ansible-builder.readthedocs.io/en/stable/definition/>`_.
+
+5. Build the EE container image called ``postgresql_ee`` (if you use docker, add the ``--container-runtime docker`` argument to the command):
+
+.. code-block:: bash
+
+  $ ansible-builder build --tag postgresql_ee
+
+6. You now can see the container image in the list of images. Assuming you use podman:
+
+.. code-block:: bash
+
+  $ podman image list
+
+  localhost/postgresql_ee          latest      2e866777269b  6 minutes ago  1.11 GB
+
+7. Inspect the ``Containerfile`` file (``Dockerfile`` if you use docker) in the ``context`` directory generated by ``ansible-builder`` to see the EE container image configuration.
+
+The created image represents an :ref:`Ansible control node<terminology>` which now contains:
+
+* Python
+* ansible-core
+* ansible-runner
+* the community.postgresql collection
+* the psycopg2-binary Python package
+
+See the :ref:`Running your EE in command line<run_firt_ee>` section to learn how to test the EE you have just created.
+
+For more information about Ansible Builder, see its :ref:`official documentation <https://ansible-builder.readthedocs.io/en/stable/>`_.
+
+
+.. _run_first_ee:
+
+Running your EE in command line
+===============================
+
+Here, we will test the EE you created in the :ref:`Building your first EE<build_first_ee>` section against the localhost and a remote target.
+
+You can now proceed being in your home (or any other arbitrary) directory.
+
+Run against localhost
+---------------------
+
+1. Create the ``test_localhost.yml`` playbook file with the following content:
+
+.. code-block:: yaml
+
+  ---
+  - hosts: localhost
+    become: yes
+    gather_facts: yes
+    tasks:
+    - name: Print facts
+      ansible.builtin.debug:
+        msg: '{{ ansible_facts }}'
+
+2. Run the playbook inside the EE container with ``ansible-navigator``:
+
+.. code-block:: bash
+
+  $ ansible-navigator run test_localhost.yml --execution-environment-image postgresql_ee --mode stdout --pull-policy missing
+
+The run will return facts gathered inside the container.
```

Comment:
Maybe......  "You may notice the facts being gathered are about the container and not the developr machine.  This is because the ansible playbook was run inside the container"

Reply:
done, thanks